tests/util/git_harness.lua

```diff
@@ -55,13 +56,24 @@ function M.in_prepared_repo(cb)
     require("neogit").setup()
     vim.cmd("Neogit")
     a.util.block_on(status.reset)
-    local _, err = pcall(cb, dir)
-    if err ~= nil then
-      error(err)
-    end
+    a.util.block_on(function()
+      local _, err = pcall(cb, dir)
+      if err ~= nil then
+        error(err)
+      end
+    end)
   end
 end
 
+---@param cmd string[]
+---@return string[]
+local function exec(cmd)
+  local result = vim.system(cmd, { text = true }):wait()
```

Comment:
It'll be cool to start using `vim.system` everywhere. I suppose we should put it somewhere that we require nightly, yea?

Reply:
Yes, most plugin I've seen require nightly or at least a recent enough version of neovim. I think I've only seen one or two plugins that do fallbacks (heirline with `nvim_get_hl` is one of them).

This is only for testing and I figured that if you are developing neovim you are using nightly, as you'd want to use apis which you know won't be replaced in the next version, like `nvim_buf_get_option` etc.

Reply:
Oh, I'm 100% in favor of nightly - I even accidentally used that new `nvim_get_hl` without realizing it was new. Oh well. We still label this project as work-in-progress, so I'm happy to lean into that for now.

Reply:
Wait wait wait! I was just browsing, but just now noticed this discussion.

It's ok to use `nightly` functionality? We are only interested in `nightly` support?

If so, this makes life *much* easier.

Reply:
Hahaha, I mean, it wasn't an intentional thing - I sorta goofed but then nobody seemed to complain. I'd still, for the most part, like to avoid things that aren't in a stable release yet, as anything in nightly is subject to change without notice. So, it's kind of a case-by-case thing. I'd definitely be open to considering using new API's.

Reply:
Damn, I was getting super hyped about going completely to nightly functions. Raining on my parade 😠.

I know that in nightly a *ton* of stuff is deprecated or just straight up removed. So it makes sense to stick to stable for most things on that note.

Reply:
Hahaha, yeah, sorry about that ⛈️  Once it hits stable then aggressive refactoring are most welcome ;)